### ISSUE TITLE
Older versions of git no longer supported

### ISSUE THREAD

**Issue by JohnLyman** [pre-commit/pre-commit#723]
Since #714 I am no longer able to run `pre-commit` on RHEL7 hosts.  The version of `git` provided by the OS vendor does not support the `-C` option.

Was this intentional?  If not, would you consider re-adding support for older versions?

My environment:
```
$ cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.3 (Maipo)

$ git --version
git version 1.8.3.1
```
I understand that version of `git` is 5+ years old, but it is the latest version provided by RedHat.  If nothing else, the documentation should state which version of `git` is required.

Thanks

**Comment by asottile** [comment]
Ooh bummer, didn't intend to remove old git support.  I'll see how feasible it is to restore that.

**Comment by asottile** [comment]
fwiw, `-C` was added in git 1.9, it's not much work to add back 1.8 support (though I'm not sure how I'll test it) -- let me prototype what that looks like :)

**Comment by asottile** [comment]
#724 to address this  :)

**Comment by asottile** [comment]
Actually, seems 1.8.5.6 (which I was using) seems to support `-C` as well, let me try 1.8.3.1 (as stated in the issue)

**Comment by asottile** [comment]
Naughty! They added this feature in a patch bump:

```
$ git describe 44e1e4d67d5148c245db362cc48c3cc6c2ec82ca
v1.8.4-1-g44e1e4d
```

**Comment by asottile** [comment]
For searchability, here's what the stacktrace looks like for this failure mode:

```
Traceback (most recent call last):
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/error_handler.py", line 47, in error_handler
    yield
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/main.py", line 259, in main
    return run(runner, args)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/commands/run.py", line 256, in run
    for _, hook in repo.hooks:
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/cached_property.py", line 32, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/repository.py", line 161, in hooks
    if hook['id'] not in self.manifest_hooks:
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/cached_property.py", line 32, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/repository.py", line 154, in manifest_hooks
    repo_path = self.store.clone(repo, rev)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/store.py", line 153, in clone
    return self._new_repo(repo, ref, deps, clone_strategy)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/store.py", line 131, in _new_repo
    make_strategy(directory)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/store.py", line 149, in clone_strategy
    _git_cmd('clone', '--no-checkout', repo, '.')
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/store.py", line 147, in _git_cmd
    return cmd_output('git', '-C', directory, *args, env=env)
  File "/tmp/pre-commit-hooks/venv/local/lib/python2.7/site-packages/pre_commit/util.py", line 178, in cmd_output
    returncode, cmd, retcode, output=(stdout, stderr),
CalledProcessError: Command: ('/tmp/git/git', '-C', '/home/asottile/.cache/pre-commit/repoKEwbEK', 'clone', '--no-checkout', 'https://github.com/asottile/reorder_python_imports', '.')
Return code: 129
Expected return code: 0
Output: (none)
Errors: 
    Unknown option: -C
    usage: git [--version] [--help] [-c name=value]
               [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
               [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
               [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
               <command> [<args>]
```

**Comment by asottile** [comment]
This was released as part of [v1.8.1](https://github.com/pre-commit/pre-commit/releases/tag/v1.8.1)!

Thanks again for the issue, sorry for the breakage!

**Comment by JohnLyman** [comment]
Great.  Thanks for the quick response!
